pkg/controllers/controllers.go

```diff
@@ -67,15 +71,19 @@ func NewControllers(
 	cloudProvider cloudprovider.CloudProvider,
 	cluster *state.Cluster,
 ) []controller.Controller {
-	p := provisioning.NewProvisioner(kubeClient, recorder, cloudProvider, cluster, clock)
+	provisioningController := dynamicprovisioning.NewProvisioningController(kubeClient, recorder, cloudProvider, cluster, clock)
+	provisioner := provisioning.NewProvisioner(kubeClient, recorder, cluster)
+
 	evictionQueue := terminator.NewQueue(kubeClient, recorder)
-	disruptionQueue := disruption.NewQueue(kubeClient, recorder, cluster, clock, p)
+	disruptionQueue := disruption.NewQueue(kubeClient, recorder, cluster, clock, provisioner)
 
 	controllers := []controller.Controller{
-		p, evictionQueue, disruptionQueue,
-		disruption.NewController(clock, kubeClient, p, cloudProvider, recorder, cluster, disruptionQueue),
-		provisioning.NewPodController(kubeClient, p, cluster),
-		provisioning.NewNodeController(kubeClient, p),
+		provisioningController, evictionQueue, disruptionQueue,
+		disruption.NewController(clock, kubeClient, provisioningController, cloudProvider, recorder, cluster, disruptionQueue),
+		dynamicprovisioning.NewPodTriggerController(kubeClient, provisioningController, cluster),
```

Comment:
Why change the names of dynamic provisioners?

Reply:
Now that we separated out NodeClaim provisioning logic to provisioner, we would have 3 controllers, actual dynamic provisioning controller that decides if it needs to create Nodeclaims for pods and trigger controllers that trigger the dynamic provsioning. Named it accordingly